common/static/common/js/discussion/utils.js

```diff
@@ -321,7 +321,8 @@
             imageUploadUrl = this.urlFor('upload');
             _processor = function(self) {
                 return function(text) {
-                    return self.postMathJaxProcessor(text);
+                    // HTML returned by Markdown is assumed to be safe to render
+                    return self.postMathJaxProcessor(edx.HtmlUtils.HTML(text)).toString();
```

Comment:
@bjacobel and @robrap explained the problem. There were actually two issues:
1. ``postMathJaxProcessor`` was changed to receive an ``HtmlSnippet``, so the parameter had to be fixed.
2. ``postMathJaxProcessor`` now returns an ``HtmlSnippet``, so it had to be converted back to a string before being returned.